src/Feature/Layout/Feature_Layout.re

```diff
@@ -0,0 +1,284 @@
+open Utility;
+
+type direction =
+  | Up
+  | Left
+  | Down
+  | Right;
+
+type t('content) =
+  | Split([ | `Horizontal | `Vertical], list(t('content)))
+  | Window({
+      weight: float,
+      content: 'content,
+    })
+  | Empty;
+
+[@deriving show({with_path: false})]
+type window('content) = {
+  content: 'content,
+  x: int,
+  y: int,
+  width: int,
+  height: int,
+};
+
+module Internal = {
+  let intersects = (x, y, split) => {
+    x >= split.x
+    && x <= split.x
+    + split.width
+    && y >= split.y
+    && y <= split.y
+    + split.height;
+  };
+
+  let move = (content, dirX, dirY, splits) => {
+    let (minX, minY, maxX, maxY, deltaX, deltaY) =
+      List.fold_left(
+        (prev, cur) => {
+          let (minX, minY, maxX, maxY, deltaX, deltaY) = prev;
+
+          let newMinX = cur.x < minX ? cur.x : minX;
+          let newMinY = cur.y < minY ? cur.y : minY;
+          let newMaxX = cur.x + cur.width > maxX ? cur.x + cur.width : maxX;
+          let newMaxY = cur.y + cur.height > maxY ? cur.y + cur.height : maxY;
+          let newDeltaX = cur.width / 2 < deltaX ? cur.width / 2 : deltaX;
+          let newDeltaY = cur.height / 2 < deltaY ? cur.height / 2 : deltaY;
+
+          (newMinX, newMinY, newMaxX, newMaxY, newDeltaX, newDeltaY);
+        },
+        (0, 0, 1, 1, 100, 100),
```

Comment:
Where do these `100` values come from?

Reply:
It seems like these are the delta - what if I'm moving from a very large window (ie, `800`) past a smaller window (ie, `50`) - it seems like this `move` could overshoot small windows. Might've been a mistake I made in the logic

Reply:
I haven't touched this part of the code, though I really, really wanted to because it's so imperative. But it seems to work fine, so "don't fix what ain't broke" and such. I'll probably do a pass over this when implementing the resizing though, because it'll need to account for the weights and locked windows and such then.